stdlib/public/core/Sequence.swift

```diff
@@ -1126,6 +1147,31 @@ extension Sequence where
     }
     return AnySequence(result)
   }
+  
+  /// Returns a subsequence by skipping elements while `predicate` returns
+  /// `true` and returning the remaining elements.
+  ///
+  /// - Parameter while: A closure that takes an element of the
+  ///   sequence as its argument and returns `true` if the element should
+  ///		be skipped or `false` if it should be included. Once the predicate
+  ///		returns `false` it will not be called again.
+  public func drop(
+    while predicate: @noescape (Iterator.Element) throws -> Bool
+  ) rethrows -> AnySequence<Iterator.Element> {
+    var iterator = makeIterator()
+    var nextElement: Iterator.Element? = iterator.next()
+    repeat {
+      nextElement = iterator.next()
+    } while try nextElement.flatMap(predicate) == true
+
+    return AnySequence {
+      AnyIterator {
```

Comment:
This is a simple implementation, but not a very efficient one.  We should use a custom type for the inner sequence.  We can reuse `_DropFirstSequence` used in `dropFirst`, by skipping the first few elements, and then wrapping the iterator in a `_DropFirstSequence`.

We should also provide specialized implementations of `drop(while:)` on `_DropFirstSequence`, so that when the user repeatedly calls `drop(while:)` we don't add levels of wrappers.  Please take a look at `_DropFirstSequence.dropFirst()`.